replay/replayapp.go

```diff
@@ -0,0 +1,140 @@
+package replay
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/loomnetwork/go-loom/builtin/types/chainconfig"
+	"github.com/loomnetwork/go-loom/types"
+	"github.com/pkg/errors"
+	abci "github.com/tendermint/tendermint/abci/types"
+	ttypes "github.com/tendermint/tendermint/types"
+
+	"github.com/loomnetwork/loomchain/state"
+	"github.com/loomnetwork/loomchain/store"
+	"github.com/loomnetwork/loomchain/txhandler"
+)
+
+type ReplayApplication interface {
+	abci.Application
+	SetTracer(vm.Tracer) error
+}
+
+var _ abci.Application = (*replayApplication)(nil)
+var _ abci.Application = ReplayApplication(nil)
+
+type replayApplication struct {
```

Comment:
This shouldn't exist, you should replay with a new instance of the `Application` that has the stores swapped out to in-memory ones, and mock event handlers / subscriptions. We can't just use a mock app for the replay, because we need all the state changing code in BeginBlock/EndBlock/Commit to run just like it did originally.